src/reports.py

```diff
@@ -78,7 +87,12 @@ def date_range(self) -> tuple[datetime.date, datetime.date]:
         """
         Inclusive date range for this column.
         """
-        start_date = self._start_date() + datetime.timedelta(weeks=self.week)
+        start_date = self._start_date()
+        current_week = 0
+        while current_week < self.week:
+            if not self._is_break(start_date):
+                current_week += 1
+            start_date += datetime.timedelta(days=7)
```

Comment:
I think this approach is a good start, but might have a bit of an issue. It appears this implementation completely ignores break weeks (ie, it pretends break weeks don't exist). This isn't really what we want, as we want people to be able to submit reports during break weeks, but we just don't want to make them required.

You probably don't even need to change anything in the calculation of weeks. You just need to change the behavior of the reports system:
1. The reports submission button should display "Submit your report! (optional due to break)" ([relevant code](https://github.com/uf-mil/discord-bot/blob/main/src/reports.py#L536-L557))
2. Only the first individual reminder should go off, so the other two will need to check whether we are currently in a break week and then disable their behavior.
3. Rather than sending out the grading form to leaders, everyone should be automatically graded green. ([relevant function](https://github.com/uf-mil/discord-bot/blob/main/src/reports.py#L838))